command/agent/job_endpoint.go

```diff
@@ -401,6 +406,91 @@ func (s *HTTPServer) jobRunAction(resp http.ResponseWriter, req *http.Request, j
 	return s.execStream(conn, &args)
 }
 
+func (s *HTTPServer) jobTagVersion(resp http.ResponseWriter, req *http.Request, jobID string, name string) (interface{}, error) {
+	switch req.Method {
+
+	// TODO: Request for reviewers: I am splitting by method here, but you'll notice both
+	// methods now call the same RPC method, which later splits them based on presence of args.Tag.
+	// So far the benefit of a method-based split here is that I can more easily create a slimmed-down
+	// TagVersionRequest for the RPC method, but I'm not sure if that's enough to justify the split.
+	// ...is this the best and most obvious place to set the .Tag or not?
+	case http.MethodPut, http.MethodPost:
+		return s.jobVersionApplyTag(resp, req, jobID, name)
+	case http.MethodDelete:
+		return s.jobVersionUnsetTag(resp, req, jobID, name)
+	default:
+		return nil, CodedError(405, ErrInvalidMethod)
+	}
+}
+
+// TODO: this is a copy of the function in command/job_history.go; any way to import it here?
```

Comment:
That'd be a circular reference as `command` imports `command/agent`. You could make this function exported and then have `command` call it as `agent.ParseVersion`, but this code seems tiny enough that it's not worth bothering.